lib/components/artworks/grid.js

```diff
@@ -0,0 +1,221 @@
+      const aspectRatio = artwork.image.aspect_ratio || 1; // Ensure we never divide by null/0
+      // Invert the aspect ratio so that a lower value means a shorter section.
+      sectionRatioSums[sectionIndex] += (1 / aspectRatio);
+    }
+
+    return sectionedArtworks;
+  }
+
+  renderSections() {
+    const spacerStyle = {
+      height: this.props.itemMargin,
+    };
+
+    const sectionedArtworks = this.sectionedArtworks();
+    const sections = [];
+    for (let i = 0; i < this.props.sectionCount; i++) {
+      const artworkComponents = [];
+      const artworks = sectionedArtworks[i];
+      for (let j = 0; j < artworks.length; j++) {
+        artworkComponents.push(<Artwork artwork={artworks[j]} key={'artwork-'+j} />);
+        // Setting a marginBottom on the artwork component didn’t work, so using a spacer view instead.
+        if (j < artworks.length-1) {
+          artworkComponents.push(<View style={spacerStyle} key={'spacer-'+j} accessibilityLabel='Spacer View' />);
+        }
+      }
+
+      const sectionSpecificStyle = {
+        width: this.state.sectionDimension,
+        marginRight: (i == this.props.sectionCount-1 ? 0 : this.props.sectionMargin),
+      };
+      sections.push(
+        <View style={[styles.section, sectionSpecificStyle]} key={i} accessibilityLabel={'Section ' + i}>
+          {artworkComponents}
+        </View>
+      );
+    }
+    return sections;
+  }
+
+  // Lifted pretty much straight from RN’s ListView.js
+  onScroll(event) {
+    const scrollProperties = event.nativeEvent;
+    const contentLength = scrollProperties.contentSize.height;
+    if (contentLength !== this._sentEndForContentLength) {
+      const offset = scrollProperties.contentOffset.y;
+      const visibleLength = scrollProperties.layoutMeasurement.height;
+      const distanceFromEnd = contentLength - visibleLength - offset;
+      if (distanceFromEnd < 100) {
+        this._sentEndForContentLength = contentLength;
+        this.fetchNextPage();
+      }
+    }
+  }
+
+  render() {
+    const artworks = this.state.sectionDimension ? this.renderSections() : null;
+    return (
+      <ScrollView onScroll={this.props.paginate ? this.onScroll : null}
+                  scrollEventThrottle={50}
+                  onLayout={this.onLayout}
+                  accessibilityLabel='Artworks ScrollView'>
+        <View style={styles.container} accessibilityLabel='Artworks Content View'>
+          {artworks}
+        </View>
+      </ScrollView>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: 'row',
+  },
+  section: {
+    flex: 1,
+    flexDirection: 'column',
+  },
+});
+
+// TODO: While we do pagination manually, we can’t actually use a Relay container around Artwork.
+const artworksQuery = (artistID, page) => {
+  return `
+    query {
+      artist(id: "${artistID}") {
+        artworks(sort: published_at_asc, filter: IS_NOT_FOR_SALE size: 10, page: ${page}) {
```

Comment:
Does the filter exclude for sale works or make the query return only for sale works?